### PR TITLE
Add installation and environments documentation

### PR DESCRIPTION
**Pull request by karan6181**
## Description of changes:
- Added streaming installation documentation.
- Added streaming environment documentation.

## Issue #, if available:
STR-3, STR-51

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [x] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [ ] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#running-tests))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: docs/source/index.md [hunk 1]
```diff
@@ -2,39 +2,30 @@
...
+- **High throughput**: Our MDS format cuts extraneous work to the bone, resulting in ultra-low sample latency and higher throughput compared to alternatives for workloads bottlenecked by the dataloader.
+- **Equal Convergence**: Model convergence from using StreamingDataset is just as good as using local disk, thanks to our shuffling algorithm. StreamingDataset shuffles across all samples assigned to a node, whereas alternative solutions only shuffle samples in a smaller pool (within a single process).
+- **Random access**: Access the data you need when you need it. Even if a sample isn’t downloaded yet, you can access `dataset[i]` to get sample `i`.
 
-- Drop-in replacement for {class}`torch.utils.data.IterableDataset` class.
-- Built-in support for popular open source datasets (e.g., ADE20K, C4, COCO, Enwiki, ImageNet, etc.).
-- Support for various image, structured and unstructured text formats.
-- Helper utilities to convert proprietary datasets to streaming format.
-- Streaming dataset compression (e.g., gzip, snappy, zstd, bz2, etc.).
```
Comment (knighton) [inline docs/source/index.md:36]: Are we just dropping these items?

File: docs/source/getting_started/Installation.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Installation
+Streaming Dataset installs via pip. Installing Streaming Dataset in a virtual environment is recommended to avoid any dependency conflicts. Streaming Dataset has been tested on Python 3.8, 3.9, and 3.10.
```
Comment (knighton) [inline docs/source/getting_started/Installation.md:2]: Streaming Dataset here, StreamingDataset below (let's stick with class name?)

**Review by knighton — APPROVED** [review]

File: docs/source/index.md (hunk 1 above)
Comment (karan6181) [reply docs/source/index.md:36]: It is already being covered as part of [README.md](https://github.com/mosaicml/streaming/blob/main/README.md). We would like to keep it short and concise and couple of above points we will be covering as part of separate doc in detail.

File: docs/source/getting_started/Installation.md (hunk 2 above)
Comment (karan6181) [reply docs/source/getting_started/Installation.md]: Done as part of https://github.com/mosaicml/streaming/pull/221/commits/d70606de7f40a64dfa250aaf6b6c66c10757c6a6

**Pull request merged**